### DISCUSSION TITLE
Can't set config on deactivate

### DISCUSSION THREAD

**Discussion by sudoaugustin** [microsoft/vscode-discussions#35] [Extension Development QnA] [answered]
I want to run this code on my extension's deactive

`workspace.getConfiguration('$name').update('$name',$value)`

But it doesn't work. (I think coz all VSCode services are shutting down when closing VSCode window)

**Comment by alefragnani** [comment]
Hi @sudoaugustin ,

Are you using the `deactivate` function, as described in [Extension Anatomy](https://code.visualstudio.com/api/get-started/extension-anatomy#extension-entry-file)?

> The extension entry file exports two functions, `activate` and `deactivate`. `activate` is executed when your registered Activation Event happens. `deactivate` gives you a chance to clean up before your extension becomes deactivated. For many extensions, explicit cleanup may not be required, and the `deactivate` method can be removed. However, if an extension needs to perform an operation when VS Code is shutting down or the extension is disabled or uninstalled, this is the method to do so.

If you are using it as described, but it is not working/being called, maybe you could open an issue in VS Code repo.

Hope this helps

**Reply by sudoaugustin** [reply]
The `deactivate` get called. But I can't update `configs` & `globalStates`.  _A similar thing happens with`postMessages` on the `unload` event not triggered._ 

I think all vscode services(`config` and `globalState` API) are being shut down before extension `deactivate` gets called.

I can even update the `setting.json` via `fs` but not with `getConfiguration`

**Comment by gjsjohnmurray** [comment, marked as answer]
Maybe https://github.com/microsoft/vscode/issues/144118 is relevant.
